hal/serial_api.h

```diff
@@ -108,9 +108,69 @@ extern "C" {
 
 /**
  * \defgroup hal_GeneralSerial Serial Configuration Functions
+ *
+ * # Defined behaviour
```

Comment:
It would also be good to add interrupt latency requirements to defined and/or undefined behavior. 
 - Correct operation guaranteed when interrupt latency is `< 8 / baudrate_frequency` without flow control
 - Correct operation guaranteed regardless of interrupt latency with flow control

For reference USB HAL spec has some timing requirements
https://github.com/ARMmbed/mbed-os/blob/master/usb/device/USBPhy/USBPhy.h#L33-L34:
https://github.com/ARMmbed/mbed-os/blob/master/usb/device/USBPhy/USBPhy.h#L58-L61